toolkit/smart-contracts/offchain/src/d_param/mod.rs

```diff
@@ -29,8 +29,15 @@ use sidechain_domain::{DParameter, UtxoId};
 #[cfg(test)]
 mod tests;
 
+/// Trait for [upsert_d_param], to make it mockable.
 pub trait UpsertDParam {
 	#[allow(async_fn_in_trait)]
+	/// This function upserts D-param.
+	/// Arguments:
+	///  - `retries`: Configuration for the retry logic of the transaction.
+	///  - `genesis_utxo`: UTxO identifying the Partner Chain.
+	///  - `d_parameter`: [DParameter] to be upserted.
+	///  - `payment_signing_key`: Signing key of a governance authority member.
```

Comment:
Actually this key doesn't have to be related to governance authority. This key will pay the fees. If it is "the governance authority key" (single key governance), then transaction will be submitted, in other cases transaction cbor is printed out. Am I right?